addons/sale_margin/models/sale_order.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+import openerp.addons.decimal_precision as dp
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    margin = fields.Float(compute='_product_margin', digits=dp.get_precision('Product Price'), store=True)
+    purchase_price = fields.Float(string='Cost Price', digits=dp.get_precision('Product Price'))
+
+    @api.multi
+    @api.onchange('product_id', 'product_uom_qty')
+    def product_id_change_margin(self):
+        for line in self:
```

Comment:
I agree it don;t think it's be needed. That's something you should ask to @nim-odoo as he changed it at b18e78e but anyway if it was working before, should still work after. If there were bugs in the module, it should be fixed in another commit. Anyway, maybe there were reasons (e.g. manually called from outside of the onchange) so you are risking to break a flow by changing it.

When checking the diff in 2b159bd9a5495c631bec237a132ba5dfbcef2443, there is no diff for this method, which is good.